src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/IndexQueueService.kt

```diff
@@ -41,17 +41,15 @@ class IndexQueueService(
     return queueAttributes.attributes["ApproximateNumberOfMessages"]?.toInt() ?: 0
   }
 
-  fun getNumberOfMessagesCurrentlyInFlight(): Int {
-    val queueAttributes = indexAwsSqsClient.getQueueAttributes(indexQueueUrl, listOf("ApproximateNumberOfMessagesNotVisible"))
-    return queueAttributes.attributes["ApproximateNumberOfMessagesNotVisible"]?.toInt() ?: 0
-  }
+  fun getIndexQueueStatus(): IndexQueueStatus {
+    var queueAttributes = indexAwsSqsClient.getQueueAttributes(indexQueueUrl, listOf("ApproximateNumberOfMessages", "ApproximateNumberOfMessagesNotVisible")).attributes
 
-  fun getIndexQueueStatus(): IndexQueueStatus =
-    IndexQueueStatus(
-      messagesOnQueue = getNumberOfMessagesCurrentlyOnIndexQueue(),
-      messagesInFlight = getNumberOfMessagesCurrentlyInFlight(),
-      messagesOnDlq = getNumberOfMessagesCurrentlyOnIndexDLQ()
+    return IndexQueueStatus(
+      queueAttributes["ApproximateNumberOfMessages"]?.toInt() ?: 0,
+      getNumberOfMessagesCurrentlyOnIndexDLQ(),
```

Comment:
I didn't think about the fact this needs 2 calls as there are 2 queues :(  Still, it's an improvement.

Reply:
Yeah, shame we can't do it all in one go

Reply:
Hmmm I wonder if we should make the calls asynchronously?  @andymarke would like that :)